src/nnue/network.h

```diff
@@ -49,6 +51,9 @@ enum class EmbeddedNNUEType {
 
 using NetworkOutput = std::tuple<Value, Value>;
 
+// The network must be a trivial type, i.e. the memory must be in-line.
+// This is required to allow sharing the network via shared memory, as
+// there is no way to run destructors.
 template<typename Arch, typename Transformer>
```

Comment:
Should we add a `static_assert(std::is_trivially_destructible_v<...>)` somewhere? Is that not possible?